packages/ckeditor5-package-generator/lib/utils/get-package-name.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @license Copyright (c) 2020-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const chalk = require( 'chalk' );
+
+const SCOPED_PACKAGE_REGEXP = /^@([^/]+)\/ckeditor5-([^/]+)$/;
+const WORD_REGEXP = /[A-Za-z](?:[a-z]+)?/g;
+
+/**
+ * This function verifies the provided package name. If it is not valid, it logs a message and exits the process.
+ * If it is valid, returns an object with the name in the following formats:
+ *
+ *   fullScoped: @scope/ckeditor5-super-feature-name
+ *   pascalCase: SuperFeatureName
+ *    camelCase: superFeatureName
+ *    kebabCase: super-feature-name
+ *    lowerCase: superfeaturename
+ *    spacedOut: Super feature name
+ *
+ * @param {Logger} logger
+ * @param {String} fullPackageName
+ * @param {CKeditor5PackageGeneratorOptions} options
+ * @returns {PackageName}
+ */
+module.exports = function getPackageName( logger, fullPackageName, options ) {
```

Comment:
I wouldn't say I like that the `getPackageName()` function does more than it describes.

* It validates the package name (old `validatePackageName()` function) and terminates the process if the input is wrong.
* It does not return a package name as it suggests (it is an object instead of a string).
* It also verifies if the provided `name` is valid. So much stuff in a single function returning a name of a package.

Consider redesigning the flow, please:

* Restore the `validatePackageName()` function as it worked perfectly.
* Add an optional validator for verifying the provided class name.
* Create a value object that requires the full package name as the argument in the constructor.
	* Then, create methods that map the name.
	* `toPascalCase(): string` – converts the given name to pascal case.
	* etc.